reducers/photos_reducer.js

```diff
@@ -24,227 +24,150 @@ const INITIAL_STATE = {
 };
 
 export default function(state = INITIAL_STATE, action) {
-    switch (action.type) {
-        /**
-         * The user has manually taken a photo with the Camera
-         */
-        case ADD_PHOTO:
-            let photos1 = [...state.photos];
-
-            let id = 0;
-            if (photos1.length > 0) id = photos1[photos1.length - 1].id + 1;
-
-            photos1.push({
-                id,
-                lat: action.payload.lat,
-                lon: action.payload.lon,
-                uri: action.payload.result.uri,
-                filename: action.payload.filename,
-                date: action.payload.date,
-                type: 'camera', // Photos taken from the camera
-                selected: false,
-                tags: {},
-                picked_up: false
-            });
-
-            return {
-                ...state,
-                photos: photos1
-            };
-
-        /**
-         * Add or update tags object on a gallery image
-         */
-
-        case ADD_TAGS_TO_CAMERA_PHOTO:
-            let photos = [...state.photos];
-            let image = photos[action.payload.currentIndex];
-
-            // update tags on image
-            let newTags = { ...image.tags };
-
-            let quantity = 1;
-            // if quantity exists, assign it
-            if (action.payload.tag.hasOwnProperty('quantity')) {
-                quantity = action.payload.tag.quantity;
-            }
-
-            // Increment quantity from the text filter
-            // sometimes (when tag is being added from text-filter, quantity does not exist
-            // we check to see if it exists on the object, if so, we can increment it
-            let payloadCategory = action.payload.tag.category;
-            let payloadTitle = action.payload.tag.title;
-            let quantityChanged = action.payload.quantityChanged
-                ? action.payload.quantityChanged
-                : false;
-
-            // check if category of incoming payload already exist in image tags
-            if (newTags.hasOwnProperty(payloadCategory)) {
-                // check if title of incoming payload already exist
-                if (newTags[payloadCategory].hasOwnProperty(payloadTitle)) {
-                    quantity = newTags[payloadCategory][payloadTitle];
-
-                    if (quantityChanged) {
-                        quantity = action.payload.tag.quantity;
-                    } else {
-                        quantity++;
-                    }
+    return produce(state, draft => {
+        switch (action.type) {
+            /**
+             * The user has manually taken a photo with the Camera
+             */
+            case ADD_PHOTO:
+                draft.photos.push({
+                    id: draft.photos.length > 0 ? draft.photos.length - 1 : 0,
```

Comment:
Much easier to check for `=== 0` during an operation instead of having to think about it